### PR TITLE
Addition of a gnu lua file for hercules

### LINKED ISSUE #1340: Add a lua file for gnu/spack-stack 1.7 on Hercules
Mostly what the title says. This will allow:
1. comparing the memory footprint of the variational apps against what we get when using intel.
2. testing libtorch and py-torch

### PR DESCRIPTION
**Pull request by guillaumevernieres**
- fixes #1340 

I had to comment out `da-utils` in the bundle, but other than that the system builds.

### CODE REVIEW

**Comment by CoryMartin-NOAA** [comment]
@guillaumevernieres what was the issue with da-utils?

**Review by CoryMartin-NOAA — DISMISSED** [review]

File: modulefiles/GDAS/hercules.gnu.lua [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+setenv("CXX","mpicxx")
+local mpiexec = '/opt/slurm/bin/srun'
+local mpinproc = '-n'
+setenv('MPIEXEC_EXEC', mpiexec)
+setenv('MPIEXEC_NPROC', mpinproc)
+
+setenv("CRTM_FIX","/work2/noaa/da/role-da/GDASApp/fix/crtm/2.4.0")
+setenv("GDASAPP_TESTDATA","/work2/noaa/da/role-da/GDASApp/testdata")
+setenv("GDASAPP_UNIT_TEST_DATA_PATH", "/work2/noaa/da/role-da/GDASApp/unittestdata")
+prepend_path("PATH","/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/prod_util/1.2.2/bin")
```
Comment (CoryMartin-NOAA) [inline modulefiles/GDAS/hercules.gnu.lua]: probably shouldn't be prepending this PATH at all, but especially when using GNU

**Comment by guillaumevernieres** [comment]
> @guillaumevernieres what was the issue with da-utils?

preprocessor stuff? I have enough distraction as is, so I just commented out and ignored! 

```
/work/noaa/da/gvernier/sandboxes/global-workflow/sorc/gdas.cd/bundle/da-utils/src/regridStates/fv3_grid.F90:126:83:

  126 |      if(ESMF_logFoundError(rcToCheck=ierr,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
      |                                                                                   1
Error: Unterminated character constant beginning at (1)
```

**Review by CoryMartin-NOAA — APPROVED** [review]

**Comment by CoryMartin-NOAA** [comment]
ok @guillaumevernieres no problem, we can revisit this later if we need to

**Pull request merged**